stellar-common/src/utils.rs

```diff
@@ -69,6 +69,11 @@ pub fn decode_transaction_meta(result_meta_xdr: &Vec<u8>) -> Result<TransactionM
     let buf = Cursor::new(result_meta_xdr);
     let transaction_meta = TransactionMeta::read_xdr(&mut Limited::new(buf, Limits::none()));
     match transaction_meta {
+        // You should be aware that MetaV3 was introduced as part of Protocol
+        // 20, so any ledgers that you are replaying from before that upgrade
+        // date will have older TransactionMeta versions. This principle
+        // generally goes for anywhere you may see a "Vx" varation of a
+        // structure.
```

Comment:
Does this mean that we have to add the arms for each Vx and decode each of them that way to make sure we lose no transactions? I thought the Vx were forward compatible.

Reply:
Considering we only have V3 data in the Firehose Blocks, we will only be decoding `TransactionMetaV3` for the moment. Once we have the archive nodes, then we can tackle that.